src/main/java/com/google/sps/servlets/EventServlet.java

```diff
@@ -38,19 +38,25 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String requestEndTime = request.getParameter(Constants.END_TIME_PARAM);
     String timeZone = request.getParameter(Constants.TIME_ZONE_PARAM);
 
-    //create dates from input
-    Date startDate = createDateTime(requestStartDate.substring(0, 4),
+    //create dates with times from input for time verification
+    Date startDateTime = createDateTime(requestStartDate.substring(0, 4),
           requestStartDate.substring(5, 7), requestStartDate.substring(8), 
           requestStartTime, timeZone); 
-    Date endDate = createDateTime(requestEndDate.substring(0, 4), 
+    Date endDateTime = createDateTime(requestEndDate.substring(0, 4), 
```

Comment:
All these numbers make the code difficult to read and parse. E.g. I have no idea whether this code is correct. 

Also, it seems the exact sequence of code is repeated below.

Suggestion:

Create a method called `parseDate` (or something more appropriate), and move all this code inside it.
Within the method, use an example date to document how this date is being parsed.